src/dbnode/storage/shard.go

```diff
@@ -1878,6 +1878,29 @@ func (s *dbShard) Bootstrap(
 		s.setFlushStateColdVersion(at, info.VolumeIndex)
 	}
 
+	// May be nil depending on the caching policy.
+	if retrieverMgr := s.opts.DatabaseBlockRetrieverManager(); retrieverMgr!= nil {
+		// Attempt to call CacheShardIndices now that all the shards are bootstrapped. Cannot call
+		// this earlier as block lease verification will fail due to the shards not being bootstrapped
+		// (and as a result no leases can be verified since the flush state is not yet known).
+		retriever, err := retrieverMgr.Retriever(s.namespace)
+		if err != nil {
+			multiErr = multiErr.Add(err)
+		} else {
+			s.logger.Debug("Caching shard indices",
+				zap.Uint32("shard", s.ID()))
+			if err := retriever.CacheShardIndices([]uint32{s.ID()}); err != nil {
+				multiErr = multiErr.Add(err)
+				s.logger.Error("Caching shard indices error",
+					zap.Uint32("shard", s.ID()),
+					zap.Error(err))
+			} else {
+				s.logger.Debug("Caching shard indices completed successfully",
```

Comment:
Super nit: Can we change all these log statements to start with lower capitalized letters (the whole effective go naming stuff)?